### PR TITLE
fix: update timeline view to show dynamic wf internals on first render

### PR DESCRIPTION
**Pull request by olga-union**
Signed-off-by: Olga Nad <olga@union.ai>
# TL;DR
Fixed the timeline view's first render issue with dynamic workflows, when it didn't show toggles for collapsed internal tasks without rendering the graph view.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Demo
https://user-images.githubusercontent.com/101579322/184722647-9f4847de-6e09-4dbc-a440-3db87ce4e6de.mov

## Tracking Issue
https://github.com/flyteorg/flyteconsole/issues/555

## Follow-up issue
More refactoring of `ExecutionTimeline` and `ExecutionNodeViews` coming.

### CODE REVIEW

File: packages/zapp/console/src/components/common/utils.ts [hunk 1]
```diff
@@ -16,3 +16,34 @@ export function measureText(fontDefinition: string, text: string) {
...
+
+/**
+ * Note:
+ *  Dynamic nodes are deteremined at runtime and thus do not come
+ *  down as part of the workflow closure. We can detect and place
+ *  dynamic nodes by finding orphan execution id's and then mapping
+ *  those executions into the dag by using the executions 'uniqueParentId'
+ *  to render that node as a subworkflow
+ */
+export const checkForDynamicExeuctions = (allExecutions, staticExecutions) => {
```
Comment (ursucarina) [inline packages/zapp/console/src/components/common/utils.ts:28]: extreme nit: spelling

File: packages/zapp/console/src/components/common/utils.ts [hunk 2]
```diff
@@ -16,3 +16,34 @@ export function measureText(fontDefinition: string, text: string) {
...
+  const parentsToFetch = {};
+  for (const executionId in allExecutions) {
+    if (!staticExecutions[executionId]) {
+      const dynamicExecution = allExecutions[executionId];
+      const dynamicExecutionId = dynamicExecution.metadata.specNodeId || dynamicExecution.id;
+      const uniqueParentId = dynamicExecution.fromUniqueParentId;
+      if (uniqueParentId) {
```
Comment (ursucarina) [inline packages/zapp/console/src/components/common/utils.ts:35]: nit: is the if check necessary?

**Review by ursucarina — APPROVED** [review]

File: packages/zapp/console/src/components/common/utils.ts (hunk 2 above)
Comment (olga-union) [reply packages/zapp/console/src/components/common/utils.ts:35]: yes, it's an optional field

**Review by ursucarina — APPROVED** [review]
